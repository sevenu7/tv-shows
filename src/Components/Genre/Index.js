import React from "react";
import Show from "../Show";
import { Row, Badge } from "react-bootstrap";
import { Link } from "react-router-dom";
import dummyImage from "../../assets/no-img-portrait-text.png";
import { SEE_MORE } from "../../constants";

function Genre({ type, shows }) {
  let filteredShows;
  let showList;
  if (shows) {
    filteredShows = shows.filter((show) => show.genres.includes(type));
  }
  if (filteredShows) {
    filteredShows.sort((a, b) => b.rating.average - a.rating.average); // sorting the shows according to rating(descending)
    showList = filteredShows
      .map((show) => (
        <Show
          image={show.image ? show.image.medium : dummyImage}
          name={show.name}
          show={show}
          key={`Show-${show.id}`}
        />
      ))
      .slice(0, 6); // showing maximum 6 shows per genre in dashboard
  }
  const genrePath = {
    pathname: `/genre/${type}`,
    shows: filteredShows,
    type,
  };
  return type ? (
    <>
      <h2>
        {type}
        {filteredShows.length > 6 ? ( // removing see more if shows less than or equal to 6
          <Link to={genrePath}>
            <Badge pill variant="dark" style={{ float: "right" }}>
              {SEE_MORE}
            </Badge>
          </Link>
        ) : null}
      </h2>
      <Row>{showList}</Row>
    </>
  ) : null;
}

export default Genre;
