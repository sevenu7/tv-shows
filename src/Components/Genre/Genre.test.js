import React from "react";
import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

import Genre from "./";

configure({ adapter: new Adapter() });

let wrapper;
const shows = [
  {
    id: 1,
    name: "Under the Dome",
    genres: ["Drama", "Science-Fiction", "Thriller"],
    schedule: { time: "22:00", days: ["Thursday"] },
    rating: { average: 6.5 },
    image: {
      medium:
        "http://static.tvmaze.com/uploads/images/medium_portrait/81/202627.jpg",
    },
  },
  {
    id: 2,
    name: "Under the Dome",
    genres: ["Drama", "Science-Fiction", "Thriller"],
    schedule: { time: "22:00", days: ["Thursday"] },
    rating: { average: 6.5 },
    image: {
      medium:
        "http://static.tvmaze.com/uploads/images/medium_portrait/81/202627.jpg",
    },
  },
  {
    id: 3,
    name: "Under the Dome",
    genres: ["Drama", "Science-Fiction", "Thriller"],
    schedule: { time: "22:00", days: ["Thursday"] },
    rating: { average: 6.5 },
    image: {
      medium:
        "http://static.tvmaze.com/uploads/images/medium_portrait/81/202627.jpg",
    },
  },
  {
    id: 4,
    name: "Under the Dome",
    genres: ["Drama", "Science-Fiction", "Thriller"],
    schedule: { time: "22:00", days: ["Thursday"] },
    rating: { average: 6.5 },
    image: {
      medium:
        "http://static.tvmaze.com/uploads/images/medium_portrait/81/202627.jpg",
    },
  },
  {
    id: 5,
    name: "Under the Dome",
    genres: ["Drama", "Science-Fiction", "Thriller"],
    schedule: { time: "22:00", days: ["Thursday"] },
    rating: { average: 6.5 },
    image: {
      medium:
        "http://static.tvmaze.com/uploads/images/medium_portrait/81/202627.jpg",
    },
  },
  {
    id: 6,
    name: "Under the Dome",
    genres: ["Drama", "Science-Fiction", "Thriller"],
    schedule: { time: "22:00", days: ["Thursday"] },
    rating: { average: 6.5 },
    image: {
      medium:
        "http://static.tvmaze.com/uploads/images/medium_portrait/81/202627.jpg",
    },
  },
];

beforeEach(() => {
  wrapper = shallow(<Genre />);
});

describe("<Genre />", () => {
  it("should match Snapshot", () => {
    expect(wrapper.debug()).toMatchSnapshot();
  });

  it("should match Snapshot with props", () => {
    wrapper = shallow(<Genre shows={shows} type="Drama" />);
    expect(wrapper.debug()).toMatchSnapshot();
  });

  it("should match Snapshot with more than 6 shows", () => {
    shows.push({
      id: 7,
      name: "Under the Dome",
      genres: ["Drama", "Science-Fiction", "Thriller"],
      schedule: { time: "22:00", days: ["Thursday"] },
      rating: { average: 6.5 },
      image: {
        medium:
          "http://static.tvmaze.com/uploads/images/medium_portrait/81/202627.jpg",
      },
    });
    wrapper = shallow(<Genre shows={shows} type="Drama" />);
    expect(wrapper.debug()).toMatchSnapshot();
  });
});
