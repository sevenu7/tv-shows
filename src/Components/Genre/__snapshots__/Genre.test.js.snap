// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Genre /> should match Snapshot 1`] = `""`;

exports[`<Genre /> should match Snapshot with  1`] = `
"<Fragment>
  <h2>
    Drama
  </h2>
  <Row noGutters={false}>
    <Show image=\\"http://static.tvmaze.com/uploads/images/medium_portrait/81/202627.jpg\\" name=\\"Under the Dome\\" show={{...}} />
    <Show image=\\"http://static.tvmaze.com/uploads/images/medium_portrait/81/202627.jpg\\" name=\\"Under the Dome\\" show={{...}} />
    <Show image=\\"http://static.tvmaze.com/uploads/images/medium_portrait/81/202627.jpg\\" name=\\"Under the Dome\\" show={{...}} />
    <Show image=\\"http://static.tvmaze.com/uploads/images/medium_portrait/81/202627.jpg\\" name=\\"Under the Dome\\" show={{...}} />
    <Show image=\\"http://static.tvmaze.com/uploads/images/medium_portrait/81/202627.jpg\\" name=\\"Under the Dome\\" show={{...}} />
    <Show image=\\"http://static.tvmaze.com/uploads/images/medium_portrait/81/202627.jpg\\" name=\\"Under the Dome\\" show={{...}} />
  </Row>
</Fragment>"
`;

exports[`<Genre /> should match Snapshot with more than 6 shows 1`] = `
"<Fragment>
  <h2>
    Drama
    <Link to={{...}}>
                
      <Badge pill={true} variant=\\"dark\\" style={{...}}>
        See More
      </Badge>
    </Link>
  </h2>
  <Row noGutters={false}>
    <Show image=\\"http://static.tvmaze.com/uploads/images/medium_portrait/81/202627.jpg\\" name=\\"Under the Dome\\" show={{...}} />
    <Show image=\\"http://static.tvmaze.com/uploads/images/medium_portrait/81/202627.jpg\\" name=\\"Under the Dome\\" show={{...}} />
    <Show image=\\"http://static.tvmaze.com/uploads/images/medium_portrait/81/202627.jpg\\" name=\\"Under the Dome\\" show={{...}} />
    <Show image=\\"http://static.tvmaze.com/uploads/images/medium_portrait/81/202627.jpg\\" name=\\"Under the Dome\\" show={{...}} />
    <Show image=\\"http://static.tvmaze.com/uploads/images/medium_portrait/81/202627.jpg\\" name=\\"Under the Dome\\" show={{...}} />
    <Show image=\\"http://static.tvmaze.com/uploads/images/medium_portrait/81/202627.jpg\\" name=\\"Under the Dome\\" show={{...}} />
  </Row>
</Fragment>"
`;

exports[`<Genre /> should match Snapshot with props 1`] = `
"<Fragment>
  <h2>
    Drama
  </h2>
  <Row noGutters={false}>
    <Show image=\\"http://static.tvmaze.com/uploads/images/medium_portrait/81/202627.jpg\\" name=\\"Under the Dome\\" show={{...}} />
    <Show image=\\"http://static.tvmaze.com/uploads/images/medium_portrait/81/202627.jpg\\" name=\\"Under the Dome\\" show={{...}} />
    <Show image=\\"http://static.tvmaze.com/uploads/images/medium_portrait/81/202627.jpg\\" name=\\"Under the Dome\\" show={{...}} />
    <Show image=\\"http://static.tvmaze.com/uploads/images/medium_portrait/81/202627.jpg\\" name=\\"Under the Dome\\" show={{...}} />
    <Show image=\\"http://static.tvmaze.com/uploads/images/medium_portrait/81/202627.jpg\\" name=\\"Under the Dome\\" show={{...}} />
    <Show image=\\"http://static.tvmaze.com/uploads/images/medium_portrait/81/202627.jpg\\" name=\\"Under the Dome\\" show={{...}} />
  </Row>
</Fragment>"
`;
